app/controllers/devise_registration_controller.rb

```diff
@@ -1,16 +1,16 @@
 class DeviseRegistrationController < Devise::RegistrationsController
   include CookiesHelper
 
+  skip_before_action :verify_authenticity_token, only: [:start]
```

Comment:
I've been having some security thoughts (see similar comment in the login controller).  I think the safe behaviour here is:

**If `params[:jwt]` is set:**

1. Skip CSRF validation
2. Validate the JWT (throwing an error if it's bad)
3. Blank the other params (so that if someone compromises a service's JWT signing key, they can't just sign people up via CSRF).

**If `params[:jwt]` is not set:**

1. Do not skip CSRF validation

Reply:
Nope, we're going to keep the cross-site stuff in the welcome controller.

In which case this controller doesn't need `skip_before_action` or `params[:jwt]`.

Reply:
I'll remove the `skip_before_action` and use of `params[:jwt]` from the login and registration controllers, keeping them only in the welcome controller.

However, I'm not sure it's possible to condition a `skip_before_action` based on a `params` value, since `params` hasn't been initialised at the point the `before_actions` are executed.